consensus/reactor.go

```diff
@@ -179,10 +228,11 @@ func (conR *ConsensusReactor) Receive(chID byte, src p2p.Peer, msgBytes []byte)
 	_, msg, err := DecodeMessage(msgBytes)
 	if err != nil {
 		conR.Logger.Error("Error decoding message", "src", src, "chId", chID, "msg", msg, "err", err, "bytes", msgBytes)
-		// TODO punish peer?
+		conR.MarkPeer(src, false, 1, p2p.Bad)
 		return
 	}
 	conR.Logger.Debug("Receive", "src", src, "chId", chID, "msg", msg)
+	conR.MarkPeer(src, true, 1, p2p.Correct)
```

Comment:
probably too ambitious to just mark good because we received an arbitrary consensus message. better if its receiving a vote or block part we needed

Reply:
Okay.